GetLimitedProfile/handler.ts

```diff
@@ -0,0 +1,154 @@
+import {
+  IResponseErrorNotFound,
+  IResponseSuccessJson,
+  ResponseErrorNotFound,
+  ResponseSuccessJson
+} from "italia-ts-commons/lib/responses";
+import { FiscalCode } from "italia-ts-commons/lib/strings";
+
+/**
+ * Type of a GetLimitedProfile handler.
+ *
+ * GetLimitedProfile expects a FiscalCode as input and returns a LimitedProfile or a NotFound error.
+ */
+type IGetLimitedProfileHandler = (
+  apiAuthorization: IAzureApiAuthorization,
+  clientIp: ClientIp,
+  userAttributes: IAzureUserAttributes,
+  fiscalCode: FiscalCode
+) => Promise<
+  | IResponseSuccessJson<LimitedProfile>
+  | IResponseErrorNotFound
+  | IResponseErrorQuery
+>;
+
+/**
+ * Whether the sender service is allowed to send
+ * notififications to the user identified by this profile
+ */
+export function isSenderAllowed(
+  blockedInboxOrChannels: IProfileBlockedInboxOrChannels | undefined,
+  serviceId: ServiceId
+): boolean {
+  return (
+    blockedInboxOrChannels === undefined ||
+    blockedInboxOrChannels[serviceId] === undefined ||
+    !blockedInboxOrChannels[serviceId].has(BlockedInboxOrChannelEnum.INBOX)
+  );
+}
+
+/**
+ * Converts the Profile model to LimetedProfile type.
+ */
+export function toLimitedProfile(
+  profile: Profile,
+  senderAllowed: boolean
+): LimitedProfile {
+  return {
+    preferred_languages: profile.preferredLanguages,
+    // computed property
+    sender_allowed: senderAllowed
+  };
+}
+
+/**
+ * Returns a type safe GetLimitedProfile handler.
+ */
+export function GetLimitedProfileHandler(
+  profileModel: ProfileModel
+): IGetLimitedProfileHandler {
+  return async (_, __, userAttributes, fiscalCode) => {
+    const maybeProfileOrError = await profileModel.findOneProfileByFiscalCode(
+      fiscalCode
+    );
+    if (isRight(maybeProfileOrError)) {
+      const maybeProfile = maybeProfileOrError.value;
+      if (isSome(maybeProfile)) {
+        const profile = maybeProfile.value;
+
+        return ResponseSuccessJson(
+          toLimitedProfile(
+            profile,
+            isSenderAllowed(
+              profile.blockedInboxOrChannels,
+              userAttributes.service.serviceId
+            )
+          )
+        );
+      } else {
+        return ResponseErrorNotFound(
+          "Profile not found",
+          "The profile you requested was not found in the system."
+        );
+      }
+    } else {
+      return ResponseErrorQuery(
+        "Error while retrieving the profile",
+        maybeProfileOrError.value
+      );
+    }
+  };
+}
+
+/**
+ * Wraps a GetLimitedProfile handler inside an Express request handler.
+ */
+export function GetLimitedProfile(
+  serviceModel: ServiceModel,
+  profileModel: ProfileModel
+): express.RequestHandler {
+  const handler = GetLimitedProfileHandler(profileModel);
+
+  const middlewaresWrap = withRequestMiddlewares(
+    AzureApiAuthMiddleware(
+      new Set([UserGroup.ApiLimitedProfileRead, UserGroup.ApiFullProfileRead])
```

Comment:
since this API will be used only by 3rd party services, we can limit the groups to `ApiLimitedProfileRead` (`ApiFullProfileRead` is only used by the app backend)

```suggestion
      new Set([UserGroup.ApiLimitedProfileRead])
```